server/datastore/mysql/hosts.go

```diff
@@ -522,6 +529,24 @@ func filterHostsByTeam(sql string, opt fleet.HostListOptions, params []interface
 	return sql, params
 }
 
+func filterHostsByMDM(sql string, opt fleet.HostListOptions, params []interface{}) (string, []interface{}) {
+	if opt.MDMIDFilter != nil {
+		sql += ` AND hmdm.mdm_id = ?`
+		params = append(params, *opt.MDMIDFilter)
+	}
+	if opt.MDMEnrollmentStatusFilter != "" {
+		switch opt.MDMEnrollmentStatusFilter {
+		case fleet.MDMEnrollStatusAutomatic:
+			sql += ` AND hmdm.enrolled IS TRUE AND hmdm.installed_from_dep IS TRUE`
```

Comment:
`enrolled IS TRUE` will trigger a full table scan, you might want to rewrite this as `enrolled = 1 AND installed_from_dep = 1` (same goes for the other statements)

Reply:
Wow, thanks! Did not know that, verified the explain plan and indeed that's the behaviour. I found it weird at first, but given that there's no boolean type _per se_, just a tinyint, it makes sense, everything that isn't 0 is true so it cannot use an index on an undefined value.